src/ALZ/Private/Config-Helpers/Write-TfvarsJsonFile.ps1

```diff
@@ -0,0 +1,49 @@
+function Write-TfvarsJsonFile {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $tfvarsFilePath,
+
+        [Parameter(Mandatory = $false)]
+        [PSObject] $configuration
+    )
+
+    if ($PSCmdlet.ShouldProcess("Download Terraform Tools", "modify")) {
+
+        if(Test-Path $tfvarsFilePath) {
+            Remove-Item -Path $tfvarsFilePath
+        }
+
+        $jsonObject = @{}
+
+        foreach($configurationProperty in $configuration.PSObject.Properties) {
+            $configurationValue = $configurationProperty.Value.Value
+
+            if($configurationProperty.Value.Validator -eq "configuration_file_path") {
+                $configurationValue = [System.IO.Path]::GetFileName($configurationValue)
+            }
+
+            if($configurationProperty.Value.DataType -eq "list(string)" -and !($configurationValue -is [array])) {
+                if($configurationValue -eq "") {
+                    $configurationValue = @()
+                } else {
+                    $configurationValue = @($configurationValue -split ",")
+                }
+            }
+
+            if($configurationProperty.Value.DataType -eq "number") {
+                $configurationValue = [int]($configurationValue)
+            }
+
+            if($configurationProperty.Value.DataType -eq "bool") {
+                $configurationValue = [bool]($configurationValue)
+            }
+

```

Comment:
If we are switching to this function, can we add map(string) and list(object) as I added in `Write-TfvarsFile` as this was a request from another team.

Also, we should be able to remove `Write-TfvarsFile.ps1` since it's no longer referenced anywhere.

Reply:
Are you able to provide me more details on the use case here? There may be a better workaround using this json version. The json method can support any object type passed by the inputs file, that is why I have moved to json instead of HCL since there is no HCL serialisation tool available in PowerShell. The need for these special types is only used by the UI inputs, hence why I never implemented any complex types. So if you are passing these complex types via the config.yaml or config.json they should already work with this new version.

I don't think we should be attempting to pass these complex types via the UI. A comma separated list is about as complex as we can get. I also think we should consider deprecating the UI input method eventually too since our preferred input method is the config file now and the UI code could be cleaned up to simplify stuff.

I will delete the HCL version though, I did intend to do that and then forgot! :)

Reply:
I have made a further update to by pass those type conversions if supplied via the input file. This means the input file needs the proper type for list(string) i.e. an array of strings now. But I think this is better moving forward from a consistency perspective and only impacts the `apply_approvers` variable at the moment. I've updated the examples files and docs accordingly.